Src/Plugins/Security/Security.SharedStrings.txt

```diff
@@ -69,5 +69,6 @@
   $SEC101/053.GoogleServiceAccountKeyConsoleFormat="private_key":\s*"-----BEGIN PRIVATE KEY-----\\n(?s)(?<key>[^"]+)(?-s)\\n-----END PRIVATE KEY-----\\n(?s).{1,200}(?-s)"client_id"\s*:\s*"(?<account>[\w.-]+)"
   $SEC101/053.GoogleServiceAccountKeyConsoleRestFormat="name": "projects\/[\w-]+\/serviceAccounts\/[\w@.-]+\/keys\/(?<key>[^"]+)",
   $SEC101/054.AlibabaAccessKey=\b(?<account>LTAI([a-zA-Z0-9]{12}|[a-zA-Z0-9]{20}))\b(?s).{0,50}(?-s)\b(?<password>[\w]{30})\b
+  $SEC101/055.Cmdkey=cmdkey.{1,200}\/pass:\s*(?<password>[^\["'\s`%\\]{2,})
```

Comment:
>Cmdkey [](start = 14, length = 6)

you're not capturing enough data here! cmdkey allows the user to be specified (add it to the account fingerprint element) and target name (which goes into the fingerprint host slot). Please note, if you have a server name, an account and a password, you can try connecting to it! So we also need a dynamic validator.

 cmdkey /add:targetname /user:username /pass:password

Reply:
Regarding the username/account, I make a second pass in the validator to find it.  Passwords are more sensitive and I didn't want to exclude exposed passwords because we couldn't correctly parse the username.

Regarding dynamic validation, my understanding is that cmdkey is designed to store *any* kind of credentials.  Assuming we could identify a username and server, we don't know what that server is, what it's listening for, or how to connect to it.  Is it a VM?  A SQL Server database?  Etc.  I don't understand how dynamic validation is possible unless we run through every technology of which we are aware.  Thoughts?

---
In reply to: [598013129](https://github.com/microsoft/sarif-pattern-matcher/pull/317#discussion_r598013129) [](ancestors = 598013129)

Reply:
As discussed, we're merging this regex with AadUser and NetworkCredentials validators

---
In reply to: [598015790](https://github.com/microsoft/sarif-pattern-matcher/pull/317#discussion_r598015790) [](ancestors = 598015790,598013129)